jobs/mongodb_migration/src/mongodb_migration/check.py

```diff
@@ -16,8 +16,8 @@
 
 
 def get_random_oids(collection: Collection, sample_size: int) -> List[int]:
-    pipeline = [{"$project": {"pk": 1}}, {"$sample": {"size": sample_size}}]
-    return [s["pk"] for s in collection.aggregate(pipeline)]
+    pipeline = [{"$project": {"_id": 1}}, {"$sample": {"size": sample_size}}]
```

Comment:
why? I changed it to fix a mypy issue (pk is always available)

Reply:
It throws an error when running migration's validation it didn't find the pk field, did you validated running the migrations with existing data when modified to fix the mypy issue?

Reply:
hmmm, ok, so maybe I'm wrong in the rest of the migrations! We could open a new issue to check that (and add some tests with an extract of the real data - the migrations are not tested, not sure it's overkill, or on the opposite, a good idea)

Reply:
I created https://github.com/huggingface/datasets-server/issues/859.